src/components/shared/Modal.js

```diff
@@ -1,11 +1,9 @@
 // @flow
 
-import { DOM as dom, createFactory, Component, PropTypes } from "react";
+import React, { PropTypes, Component } from "react";
 import type { Children } from "react";
 import classnames from "classnames";
-import _Transition from "react-transition-group/Transition";
-const Transition = createFactory(_Transition);
-
+import Transition from "react-transition-group/Transition";
```

Comment:
Slide is a component. You can convert the default export to JSX and make this proper change.